### DISCUSSION TITLE
[Roadmap] Client v2

### DISCUSSION THREAD

**Discussion by Mister-Hope** [walinejs/waline#637] [General]
We are going to release v2 client very soon. Here is a list of changes:

## Drop Valine support in client

We have already marked some options as `@deprecated` in v1, and those options will be removed in v2.

Do not worry:

- all the options have a replacment. No feature will be lost. 
- We are only changing client options, and you can still import Valine database from leancloud to Waline and make it work directly

## New "Pure" Client

 We are going to release a new `@waline/client-pure` client. it's targeting minial size. 

While `@waline/client` based on vue, and is reactive, it will have 3 benefits:

- A bundled markdown parser to provide markdown preview
- It's reactive so you can use it on SPA, and simply update the comment area at anytime with new options.
- It provides option checking, so if you provide wrong options (e.g.: typos), the client probably won't fail on most situations due to extra validatation.

`@waline/client-pure` will have the following changes to help reduce size:

- `marked` is no longer bundled, so if you want a preview feature, you must provide a parser function using option `markdown`
- It's pure JavaScript without any framework, so it won't be reactive, whenever you want an update, you should provide new option then recall it, to preform a whole rerendering.
- It's droping config validation, so if you provide wrong options (e.g: provide a `string` instead of `function` in `uploadImage`), the client will crash and throw errors. So please make sure you provide a correct option.

## Finally

We will release it in `next` channel first, we hope you can try it and we are looking forward to you feedback. 😉

---

我们将很快发布 v2 客户端。以下是更改列表：

## 在客户端中移除 Valine 支持

我们已经在 v1 中将一些选项标记为"废弃"，这些选项将在 v2 中删除。

不要担心：

- 所有选项都有替代。不会丢失任何功能。
- 我们只是改变了客户端选项，您仍然可以从leancloud导入Valine数据库到Waline并使其直接工作

## 新的 “纯净” 客户端

 我们将发布一个新的 `@waline/client-pure` 客户端。它的目标是最小尺寸。

`@waline/client` 基于 vue，并且是响应式的，但它有 3 个好处：

- 捆绑的 Markdown 解析器，提供 Markdown  预览
- 它是响应式的，因此您可以在 SPA 上使用它，并随时使用新选项动态更新评论区。
- 它提供选项检查，因此如果您提供错误的选项（例如：拼写错误），由于额外的验证，客户端在大多数情况下可能不会失败。

`@waline/client-pure` 将进行以下更改以帮助减小大小：

- `marked` 不再捆绑，所以如果你想要预览功能，你必须使用选项 `markdown` 提供一个解析器函数
- 它是纯 JavaScript，没有任何框架，所以它不会是响应式的，每当你想要更新时，你应该提供新选项然后调用它，以执行整个 DOM 的重新渲染。
- 它删除了选项验证，因此如果您提供错误的选项（例如：在 `uploadImage` 中提供 `string` 而不是 `function`），客户端将无法工作并抛出错误。所以请确保您提供了正确的选项。

## 最后

我们会先在 `next` 频道发布，希望大家可以试用，期待您的反馈。 😉

**Comment by magic-H728** [comment]
gitee上的评论无法显示头像
